framework/include/base/MooseVariable.h

```diff
@@ -196,8 +196,16 @@ class MooseVariable : public MooseVariableBase
   const Node *& node() { return _node; }
   dof_id_type & nodalDofIndex() { return _nodal_dof_index; }
   bool isNodalDefined() { return _is_defined; }
-  const VariableValue & nodalSln() { return _nodal_u; }
-  const VariableValue & nodalSlnOld() { return _nodal_u_old; }
+  const VariableValue & nodalSln()
+  {
+    _need_nodal_u = true;
+    return _nodal_u;
+  }
+  const VariableValue & nodalSlnOld()
+  {
+    _need_nodal_u_old = true;
+    return _nodal_u_old;
+  }
```

Comment:
You are wasting our time. Creating an input file probably with some source files is not worth especially for a bug this obvious. If you really care, go take a look at 3c06af725cfaf3fac05fb44c3fb064a2c01bed31.